### ISSUE TITLE
Proyecto: Regresión: Optimización del Modelo

### ISSUE THREAD

**Issue by furrutiav** [furrutiav/lab-mds-2022#18]
*Aquí deberán mejorar del modelo de regresión al variar los algoritmos/hiperparámetros que están ocupando.*

- Generar una nueva `Pipeline` enfocada en buscar el mejor modelo usando GridSearch.
- Usar **`GridSearchCV`** o **`HalvingGridSearchCV`** para tunear hipermarámetros. La primera demorará más que la segunda pero les traerá potencialmente mejores resultados. Pueden probar también [`OptunaSearchCV`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.OptunaSearchCV.html) de la librería [`Optuna`](https://optuna.org/) , la cuál es bastante popular para buscar modelos de redes neuronales.
- Agregar técnicas de seleccion de atributos, como también usar mejores clasificadores y explorar sus hiperparámetros. 
- Probar distintos parámetros para las transformaciones de datos, seleccion de atributos, clasificadores, etc...
- Probar modelos basados en gradient boosting/bagging. **Recomendación fuerte:** Probar las librerías [`LightGMB`](https://lightgbm.readthedocs.io/en/latest/) o [`xgboost`](https://xgboost.readthedocs.io/en/stable/).
- Probar activando/descativando los procesadores de texto, de categorías, etc...
- Recuerden setear la búsqueda para optimizar la métrica que se evalua en la competencia.

Algunas notas interesantes sobre este proceso (las mismas que antes...): 

- No se les pide rendimientos cercanos al 100% de la métrica para concretar exitosamente el proyecto. Por otra parte, celebren cada progreso que obtengan.
- Hagan grillas computables: Si la grilla se va a demorar 1/3 la edad del universo en explorarse completamente, entonces achíquenla a algo que sepan que va a terminar. 
- Aprovechen el procesamiento paralelo (con `njobs`) para acelerar la búsqueda. Sin embargo, si tienen problemas con la memoria RAM, reduzca la cantidad de jobs a algo que su computador/interprete web pueda procesar.

**Al final de este proceso, seleccione el mejor modelo de regresión encontrado, prediga los target del test set de la competencia y envíelos a Codalab.**

**Comment by abeldano** [comment]
Falta GS
